libr/include/r_util/r_annotated_code.h

```diff
@@ -50,6 +51,10 @@ typedef struct r_code_annotation_t {
 		struct {
 			RSyntaxHighlightType type;
 		} syntax_highlight;
+
+		struct {
+			const char *name;
```

Comment:
In the current form this can't work properly without memory problems. What is managing the memory pointed by name?RAnnotatedCode is supposed to be self contained.

Reply:
Yep, this should be just `char *` and owned by this annotation. Then, you can create a free function for a single annotation that checks the type and frees the name when necessary and pass it in here instead of the NULL:
https://github.com/radareorg/radare2/blob/305cc00766516eb563fd4a6a9e74700aa0907629/libr/util/annotated_code.c#L12